src/lsp-server.ts

```diff
@@ -188,9 +188,10 @@ export class LspServer {
         // Setup supported features.
         const { textDocument } = clientCapabilities;
         if (textDocument) {
+            const completionCapabilities = textDocument.completion;
             this.features.codeActionDisabledSupport = textDocument.codeAction?.disabledSupport;
             this.features.definitionLinkSupport = textDocument.definition?.linkSupport && typescriptVersion.version?.gte(API.v270);
-            const completionCapabilities = textDocument.completion;
+            this.features.insertReplaceSupport = completionCapabilities?.completionItem?.insertReplaceSupport ?? false;
```

Comment:
Fine if the value is `undefined`:

```suggestion
            this.features.insertReplaceSupport = completionCapabilities?.completionItem?.insertReplaceSupport;
```